sso-kit-core/src/main/java/com/vaadin/sso/core/AbstractSingleSignOnProperties.java

```diff
@@ -0,0 +1,197 @@
+/*-
+ * Copyright (C) 2022 Vaadin Ltd
+ *
+ * This program is available under Vaadin Commercial License and Service Terms.
+ *
+ *
+ * See <https://vaadin.com/commercial-license-and-service-terms> for the full
+ * license.
+ */
+package com.vaadin.sso.core;
+
+/**
+ * Definition of configuration properties for the SSO Kit starter.
+ *
+ * @author Vaadin Ltd
+ * @since 1.0
+ */
+public abstract class AbstractSingleSignOnProperties {
+
+    /**
+     * The default login route. Unauthorized requests will be redirected to this
+     * route. If there are no views mapped to this route, Spring's default login
+     * page will be generated.
+     */
+    public static final String DEFAULT_LOGIN_ROUTE = "/login";
+
+    /**
+     * The default logout-redirect route. Web browsers will be redirected to
+     * this route after the logout process has completed.
+     */
+    public static final String DEFAULT_LOGOUT_REDIRECT_ROUTE = "{baseUrl}";
+
+    /**
+     * The default Back-Channel Logout route. This should be the same as in the
+     * OIDC provider's configuration to be able to accept logout notices as
+     * described by the specification. It requires a URI variable to match the
+     * client registration-id: {@code registrationId}.
+     *
+     * @see https://openid.net/specs/openid-connect-backchannel-1_0.html
+     */
+    public static final String DEFAULT_BACKCHANNEL_LOGOUT_ROUTE = "/logout/back-channel/{"
+            + BackChannelLogoutFilter.REGISTRATION_ID_URI_VARIABLE_NAME + "}";
+
+    /**
+     * The default maximum number of concurrent sessions allowed per user: -1
+     * means any number of concurrent sessions is allowed.
+     */
+    static final int DEFAULT_MAXIMUM_SESSIONS_PER_USER = -1;
+
+    /**
+     * Enables (or disables) auto-configuration.
+     */
+    private boolean autoConfigure = true;
```

Comment:
The Hilla starter configuration doesn't depend on auto-configuration property. Could be this moved out from abstract properties to a the vaadin properties specifically. It could be missleading.

Reply:
Shouldn't the Hilla auto-configuration also be possible to disable? This is useful e.g. in docs snippets where we don't want the kit to be enabled.

Reply:
Yes, good point. Definitely it should be possible.

Reply:
I added that!